lib/Sema/DerivedConformances.cpp

```diff
@@ -55,11 +55,17 @@ bool DerivedConformance::derivesProtocolConformance(DeclContext *DC,
                                                     NominalTypeDecl *Nominal,
                                                     ProtocolDecl *Protocol) {
   // Only known protocols can be derived.
-  auto knownProtocol = Protocol->getKnownProtocolKind();
+  const auto knownProtocol = Protocol->getKnownDerivableProtocolKind();
```

Comment:
@CodaFi or @DougGregor Could either of you sanity check this change?

Reply:
@rintaro Thanks for taking a look!